packages/appium/docs/en/guides/grid.md

```diff
@@ -2,4 +2,72 @@
 title: Appium and Selenium Grid
 ---
 
-TODO
+## Selenium Grid 4
+
+`Relay` feature in Grid 4 allows you to proxy Appium requests to an appium server instance.
+
+Please check [Relaying commands to a service endpoint that supports WebDriver](https://www.selenium.dev/documentation/grid/configuration/toml_options/#relaying-commands-to-a-service-endpoint-that-supports-webdriver) and [Selenium Grid 4 and Appium together in harmony](https://www.youtube.com/watch?v=3_aP2rsqZD0) about the configuration and for more details.
+
+## Selenium Grid 3
+
+You are able to register your appium server with a local [Selenium Grid 3](https://www.selenium.dev/documentation/legacy/selenium_3/grid_3/)
+([setup docs](https://www.selenium.dev/documentation/legacy/grid_3/setting_up_your_own_grid/)) by using the
+`--nodeconfig` server parameter.
+
+```center
+> appium server --nodeconfig /path/to/nodeconfig.json --base-path=/wd/hub
+```
+
+In the node config file you have to define the `browserName`,
+`version` and `platform` and based on these parameters the grid
+will re-direct your test to the right device. You will also need to
+configure your **host** details and the **selenium grid** details. For
+a full list of all parameters and descriptions look
+[here](https://github.com/SeleniumHQ/selenium/wiki/Grid2)
+
+Once you start the appium server and it registers with the grid,
+you will see your device on the grid console page:
+
+`http://**\<grid-ip-adress\>**:**\<grid-port\>**/grid/console`
+
+
+A new session capabilities should have `appium:` prefix in non-standard capabilities such as
```

Comment:
Yes. Appium 1.x accepted both, W3C and MJSONWP, so users did not need to add `appium:` since then Appium 1.x accepted it as MJSONWP. But Appium 2 does not work in this case, so users should add `appium:` explicitly.

I saw like this issue report a couple of times.

Reply:
Then, the grid adds the capabilities wth `appium:` in both `desiredCapabilities` and `firstMatch` in `capabilities`, Appium 2 will pick the firstMatch's one. We should guide grid 4 as a base line, but users still reported they use grid 3.